### PR TITLE
Background Task Improvement / Log Messages

### PR DESCRIPTION
**Pull request by arifiorino**
- Background task improvement taken from #198. 
- Before background tasks would run on every workload regardless if it changed, which is inefficient.
- Also background tasks only runs if five results have been uploaded, because it fails with too few.
- Added clean_knobs function which makes sure that there are no knobs missing or unnecessary.
- Added more log messages in periodic and async tasks.

### CODE REVIEW

File: server/website/website/tasks/periodic_tasks.py [hunk 1]
```diff
@@ -18,20 +18,27 @@
                                     DummyEncoder,
                                     consolidate_columnlabels)
 from website.models import PipelineData, PipelineRun, Result, Workload
-from website.types import PipelineTaskType
+from website.types import PipelineTaskType, WorkloadStatusType
 from website.utils import DataUtil, JSONUtil
 
 # Log debug messages
 LOG = get_task_logger(__name__)
```
Comment (dvanaken) [inline server/website/website/tasks/periodic_tasks.py]: I would change all (or most) of the log messages in this file from "info" to "debug" level (keep any messages you think should be "info" level)

File: server/website/website/tasks/periodic_tasks.py [hunk 2]
```diff
@@ -40,16 +47,28 @@ def run_background_tasks():
     pipeline_run_obj = PipelineRun(start_time=now(), end_time=None)
     pipeline_run_obj.save()
 
-    for workload in unique_workloads:
+    for workload in modified_workloads:
 
         wkld_results = Result.objects.filter(workload=workload)
         if wkld_results.exists() is False:
             # delete the workload
```
Comment (dvanaken) [inline server/website/website/tasks/periodic_tasks.py]: Add a log message here that includes the name/id of the workload being deleted

**Comment by dvanaken** [comment]
@arifiorino Is this ready?

**Comment by arifiorino** [comment]
Yes, I made the changes you requested yesterday.

**Comment by dvanaken** [comment]
Can you update your branch?

**Comment by arifiorino** [comment]
I already made a new commit “Changed log messages” which shows up under this pull request. Do I need to do anything else?

**Comment by dvanaken** [comment]
Yes, your branch is no longer in sync with the cmu-db/ottertune master branch. You can use git rebase to resolve conflicts and rewrite all of your commits on top of master. Sometimes I can do this automatically but this time I can't because there's a conflicting file: server/website/website/migrations/0003_load_initial_data.py

**Comment by arifiorino** [comment]
Ok I will do that soon.

**Comment by dvanaken** [comment]
Cool thanks

**Comment by arifiorino** [comment]
I tried merging the remote branch of the cmu-db repo with my repo to see if that worked. Travis failed because the apt-get failed, not because the tests failed.

**Comment by dvanaken** [comment]
Just merged #225 which should fix this issue.

**Comment by arifiorino** [comment]
Before the centos build worked and the ubuntu build failed and now the ubuntu build worked and the centos build failed.

**Pull request merged**